runtime/common/MeasureCounts.cpp

```diff
@@ -127,31 +140,18 @@ std::vector<std::size_t> sample_result::serialize() const {
 
 void sample_result::deserialize(std::vector<std::size_t> &data) {
   std::size_t stride = 0;
+  totalShots = 0;
+
   while (stride < data.size()) {
-    auto nChars = data[stride];
-    stride++;
-    std::string name = "";
-    for (std::size_t i = 0; i < nChars; i++)
-      name += std::string(1, char(data[stride + i]));
-
-    stride += nChars;
-    std::size_t localShots = 0;
+    std::string name = extractNameFromData(data, stride);
+
     std::unordered_map<std::string, std::size_t> localCounts;
-    auto nBs = data[stride];
-    stride++;
-    for (std::size_t j = stride; j < stride + nBs * 3; j += 3) {
-      auto bitstring_as_long = data[j];
-      auto size_of_bitstring = data[j + 1];
-      auto count = data[j + 2];
-      auto bs = longToBitString(size_of_bitstring, bitstring_as_long);
-      localShots += count;
-      localCounts.insert({bs, count});
-    }
+    deserializeCounts(data, stride, localCounts);
 
     sampleResults.insert({name, ExecutionResult{localCounts, name}});
-
-    stride += nBs * 3;
-    totalShots = localShots;
+    totalShots += std::accumulate(
```

Comment:
This seems like a change in behavior. In the old code, `totalShots` was assigned based on the last time through the `while` loop whereas in the new code, `totalShots` is the sum over all times through the `while` loop. Is that intentional?